web/src/unread_ui.js

```diff
@@ -114,8 +131,9 @@ export function initialize() {
 
         hide_unread_banner();
     });
-    $("#mark_as_read_close").on("click", () => {
+    $("body").on("click", "#mark_as_read_close", () => {
         hide_unread_banner();
         user_closed_unread_banner = true;
     });
+    hide_unread_banner();
```

Comment:
I don't understand why you're calling both `update_unread_banner` and `hide_unread_banner` during the initialization process here. Was this a workaround for the previous selector approach?

Reply:
`update_unread_banner` only updates the banner that should be shown, it doesn't exactly update the visibility of the banners. During the initalization process, the banners will be visible off start and we probably want to hide it.

Reply:
OK I'm going to do this tweak to make this somewhat strange sequence of steps easy to understand:

```
diff --git a/web/src/unread_ui.js b/web/src/unread_ui.js
index d2a3a6cdbf..819d8636a7 100644
--- a/web/src/unread_ui.js
+++ b/web/src/unread_ui.js
@@ -115,7 +115,6 @@ export function should_display_bankruptcy_banner() {
 
 export function initialize() {
     update_unread_counts();
-    update_unread_banner();
     $("body").on("click", "#mark_view_read", () => {
         // Mark all messages in the current view as read.
         //
@@ -135,5 +134,11 @@ export function initialize() {
         hide_unread_banner();
         user_closed_unread_banner = true;
     });
+
+    // The combination of these functions in sequence ensures we have
+    // at least one copy of the unread banner in the DOM, invisible;
+    // this somewhat strange pattern allows our CSS to reserve space for
+    // the banner, to avoid scroll position jumps when it is shown/hidden.
+    update_unread_banner();
     hide_unread_banner();
 }
```